### PR TITLE
ci: Deduplicate images; deploy via GitHub Pages

### PR DESCRIPTION
**Pull request by michael-wisely-gravwell**
This PR addresses no issue.

This PR proposes:

- adding `dedup-links`: A tool written in Go that walks the built docs, removing duplicate images and rewriting `` links as possible
- adding support for a `VERSION_LIST_URL` Actions variable that can override the default version list location. Useful for debugging
- adding support for a `BASE_PATH` Actions variable that indicates which subpath (if any) should be considered when rewriting paths
- modifying `build-html.yml`, so that the workflow...
  - is triggered on published release instead of `workflow_dispatch`
  - deploys the built docs to GitHub Pages (still requires approval for deployment)

This PR targets `main` because it has to be there to catch relevant `push` events.

### TODO

- [x] Add a readme for [dedup-links](https://github.com/michael-wisely-gravwell/wiki/tree/dedup/_tools/dedup-links)

### CODE REVIEW

**Comment by ashnwade** [comment]
We do always build from main, but I don't think we want to build every time there is a push to main. For instance, the changelog PR and other changes from a next-* branch will be in one PR to main, but the automated PR to update hashes will be a separate PR to main. And we really only really want to build and publish once both of those PRs are in. 

Are we sure we really want to change this to trigger on `push`?

**Comment by michael-wisely-gravwell** [comment]
>Are we sure we really want to change this to trigger on push?

I do think it's more appropriate than workflow_dispatch.

Rather than triggering a run on a push to a branch, we should have it trigger on creation of a tag. We don't create the tag until all PRs are merged are we're totally done. 

It's trivial to change the `push` rule to run on `vn.n.n` tags instead of pushes to `main`

**Comment by ashnwade** [comment]
I like that idea of doing it on tags instead.

File: .github/workflows/build-html.yml [hunk 1]
```diff
@@ -16,46 +31,87 @@ jobs:
...
           # Make temporary location for build results
           export BUILD_DEST="$( mktemp --directory --tmpdir=${{ runner.temp }} )"
 
           # Fetch all remote refs
           git fetch --all
 
-          # Pull latest from main branch
-          git checkout main
-          git pull
+          # Checkut the tag that triggered this run
```
Comment (ashnwade) [inline .github/workflows/build-html.yml:55]:
```suggestion
          # Checkout the tag that triggered this run
```

**Review by ashnwade — APPROVED** [review]
LGTM

**Pull request merged**